### DISCUSSION TITLE
Clients crash due to lack of RAM

### DISCUSSION THREAD

**Discussion by carles-alemany** [ltsp/ltsp#501] [Q&A] [answered]
Hello, 

We have two servers, one with old LTSP5 and another one with LTSP19. We noticed that when we boot computers with LTSP19, they crash more often due lack of RAM than they did with LTSP5. 

It seems that LTSP19 does not have NBD Swap or I didn't set up it correctly, and I'm unable to find information about this. 

Any thoughts on how to avoid those crashes? We will increase clients' RAM from 4GB to 8GB as a last resource. 

Thank you

**Comment by alkisg** [comment, marked as answer]
Hi, see https://github.com/ltsp/ltsp/issues/7

LTSP (currently at version 21.01) only supports local swap out of the box, so if the clients have a local disk or even a USB stick with a swap partition, it'll be used.

It's easy to create swap files over NFS if you want, with an POST_INIT_SWAP parameter in ltsp.conf. I imagine they'll be a bit more stable than NBD swap. But network swapping in general was causing so many issues in LTSP5 that I didn't want to publish any related code without *a lot* of testing.

**Reply by carles-alemany** [reply]
Hi alkish,

Good to know, it makes a lot of sense. I think we'll go with a combination of old HDDs to use as local swap, along with a RAM increasement on the most demanding clients. 

Thanks for the explanation.
